R/namespace.R

```diff
@@ -16,5 +16,20 @@ namespace_imports <- function(path = find_package()) {
   data.frame(
     pkg = unlist(lapply(imports, function(x) rep(x[[1L]], length(x[[2L]])))),
     fun = unlist(lapply(imports, `[[`, 2L)),
-    stringsAsFactors = FALSE)
+    stringsAsFactors = FALSE
+  )
+}
+
+# filter namespace_imports() for S3 generics
+# this loads all imported namespaces
```

Comment:
NB this already happens in `namespace_imports()` for completely imported packages.
IMO it's worth to verify whether imported functions are actually generics, reducing the false negative rate.